kernel/aster-nix/src/net/socket/ip/stream/connected.rs

```diff
@@ -17,13 +17,19 @@ use crate::{
 pub struct ConnectedStream {
     bound_socket: Arc<AnyBoundSocket>,
     remote_endpoint: IpEndpoint,
+    new_connection: AtomicBool,
```

Comment:
This field is rather confusing to be put in `ConnectedStream`. I think this field only takes effect when nonblocking mode is set in connecting, right? Can the field be added in `ConnectingStream` instead of `ConnectedStream`?

If can't, some comments should be added to this field to explain when to set and clear the field.